src/services/orders.js

```diff
@@ -0,0 +1,97 @@
+const { Orders } = require('../models');
+const { OrderStatuses } = require('../models');
+
+class OrdersService {
+  async createOrder (req, res) {
+    try {
+      const orderStatus = await OrderStatuses.findOne({
+        name: 'In progress'
+      }).exec();
+      const order = new Orders(req.body);
+      order.status = orderStatus._id;
+      await order.save();
+      return order ? res.json(true) : res.boom.notFound();
+    } catch {
+      res.boom.badImplementation();
+    }
+  }
+
+  async getOrders (req, res) {
+    try {
+      const { _id, status } = req.query;
+      const query = {};
+
+      if (_id) query._id = _id;
+      if (status) query.status = status;
+
+      const count = Orders.find(query).count();
+      const orders = await Orders.find(query).exec();
+      return {
+        total_size: count,
+        list: orders ? res.json(orders) : res.json([])
+      };
+    } catch {
+      res.boom.badImplementation();
+    }
+  }
+
+  async updateOrder (req, res) {
+    try {
+      const statusInProgress = await OrderStatuses.findOne({
+        name: 'In progress'
+      }).exec();
+      const order = await Orders.findOne({ _id: req.params.id }).exec();
+      if (order.status.toString() === statusInProgress._id.toString()) {
+        order.products = req.body.products;
```

Comment:
When you set order.products to req.body.products, you remove the others that are already set to order.products, if any.

Reply:
I don't see any change here. You should only add or remove products to the updated order, not setting products to the current state of req.body.products.